src/mono/mono/mini/mini-exceptions.c

```diff
@@ -3254,7 +3254,7 @@ mono_thread_state_init (MonoThreadUnwindState *ctx)
 {
 	MonoThreadInfo *thread = mono_thread_info_current_unchecked ();
 
-#if defined(MONO_CROSS_COMPILE)
+#if defined(MONO_CROSS_COMPILE) || defined(HOST_WASI)
 	ctx->valid = FALSE; //A cross compiler doesn't need to suspend.
```

Comment:
Why is this needed ?

Reply:
We hit the error `Use a null sigctx requires a working mono-context` otherwise. Overall the threading mechanism within WASI needs to be integrated more properly. I've tweaked this code to have a clearer comment now, but it remains one of the "todo" items for production-quality WASI support.